base/Command.js

```diff
@@ -53,10 +53,14 @@ class Command {
   }
 
   async verifyUser(message, user, options = {}) {
-    const match = /(?:<@!?)?([0-9]{17,20})>?/gi.exec(user);
-    if (!match) throw new ParseError("Invalid Mention or ID", options.msg);
-    const id = match[1];
-    return this.client.users.fetch(id);
+    let match;
+    const idMatch = /(?:<@!?)?([0-9]{17,20})>?/gi.exec(user);
+    const nameMatch = /./gi.test(user);
+    if (!idMatch && !nameMatch) throw new ParseError("Invalid Mention or ID", options.msg);
+    if (idMatch) match = idMatch[1];
+    if (/(#[0-9]{4})$/.test(user)) match = user.substr(0, user.length-5);
+    if (!idMatch) match = message.guild.members.find(m => m.user.username === match).id;
+    return this.client.users.fetch(match);
```

Comment:
With the code given above in the other requested changes, you'd cover:

1. Attempts to parse mention or ID
    - If success, return the output from the API (or cache, if the user is cached).
1. Checks if the input includes a discriminator
    - Finds a member by its user tag.
1. Definitively a username. Finds it without discriminator
1. If either the points 2 or 3 found a match, return [`GuildMember#user`](https://discord.js.org/#/docs/main/stable/class/GuildMember?scrollTo=user).
1. Else, there's no match by mention, tag or username. Throw the ParseError.